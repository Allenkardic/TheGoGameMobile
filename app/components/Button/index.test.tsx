// import React from 'react';
// import {Pressable} from 'react-native';
// import renderer from 'react-test-renderer';
// import Button from './';

// describe('Button Component', () => {
//   it('renders correctly with default props', () => {
//     const tree = renderer
//       .create(
//         <Button onPress={() => {}} text="Click Me" isSubmitting={false} />,
//       )
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it('renders spinner when isSubmitting is true', () => {
//     const tree = renderer
//       .create(<Button onPress={() => {}} text="Submit" isSubmitting={true} />)
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it('disables the button when isSubmitting is true', () => {
//     const component = renderer.create(
//       <Button onPress={() => {}} text="Submit" isSubmitting={true} />,
//     );
//     const button = component.root.findByType(Pressable);
//     expect(button.props.disabled).toBe(true);
//   });

//   it('triggers onPress when button is pressed', () => {
//     const mockOnPress = jest.fn();
//     const component = renderer.create(
//       <Button onPress={mockOnPress} text="Click Me" isSubmitting={false} />,
//     );
//     const button = component.root.findByType(Pressable);
//     button.props.onPress();
//     expect(mockOnPress).toHaveBeenCalled();
//   });

//   it('applies custom styles', () => {
//     const customStyle = {backgroundColor: 'blue'};
//     const tree = renderer
//       .create(
//         <Button
//           onPress={() => {}}
//           text="Styled Button"
//           isSubmitting={false}
//           style={customStyle}
//         />,
//       )
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
