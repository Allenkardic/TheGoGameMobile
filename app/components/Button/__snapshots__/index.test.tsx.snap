// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component applies custom styles 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#031a61",
        "borderRadius": 30,
        "borderWidth": 2,
        "overflow": "hidden",
      },
      {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 30,
          "height": 60,
          "justifyContent": "center",
          "paddingVertical": 2,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "lineHeight": 24,
          },
          {
            "color": "#010a1a",
          },
          {
            "color": "#FFFFFF",
          },
          null,
          null,
        ]
      }
    >
      Styled Button
    </Text>
  </View>
</View>
`;

exports[`Button Component renders correctly with default props 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#031a61",
        "borderRadius": 30,
        "borderWidth": 2,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 30,
          "height": 60,
          "justifyContent": "center",
          "paddingVertical": 2,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "lineHeight": 24,
          },
          {
            "color": "#010a1a",
          },
          {
            "color": "#FFFFFF",
          },
          null,
          null,
        ]
      }
    >
      Click Me
    </Text>
  </View>
</View>
`;

exports[`Button Component renders spinner when isSubmitting is true 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#031a61",
        "borderRadius": 30,
        "borderWidth": 2,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 30,
          "height": 60,
          "justifyContent": "center",
          "paddingVertical": 2,
        },
      ]
    }
  >
    <ActivityIndicator
      color="white"
      size="small"
      style={
        [
          {
            "marginBottom": -4,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
